src/fileioc/fileioc.asm

```diff
@@ -1573,6 +1626,29 @@ util_set_offset:
 	ld	(hl), bc
 	ret
 
+util_Arc_Unarc: ;properly handle garbage collects :P
+	call	_ChkFindSym
+	push	hl
+	call	_ChkInRAM
+	pop	hl
+	jr	nz,.arc_unarc ;if the file is already in archive, we won't trigger a gc
+	call	_LoadDEInd_s
+	ld	hl,12
+	add	hl,de
+	call	_FindFreeArcSpot ;check if we will trigger a gc
+	jr	nz,.arc_unarc ;gc will not be triggered
+	call	util_pre_gc_default_handler
+util_pre_gc_handler:=$-3
+	or	a,a
+	ret	z ;exit if the handler returns false
```

Comment:
This is a fail state (the caller either failed to do what it is trying to do or will soon do bad things) which needs to be explicitly handled by the caller.  Especially if there were a global state for whether to allow garbage collection, ti_SetArchiveStatus needs to return whether it actually succeeded.

Reply:
removed gc canceling